scripts/composer/ScriptHandler.php

```diff
@@ -39,27 +39,36 @@ public static function createRequiredFiles(Event $event) {
     }
 
     // Prepare the settings file for installation.
-    if (!$fs->exists($drupalRoot . '/sites/default/settings.php') and $fs->exists($drupalRoot . '/sites/default/default.settings.php')) {
+    if ($fs->exists($drupalRoot . '/sites/default/default.settings.php') && !$fs->exists($drupalRoot . '/sites/default/settings.php')) {
```

Comment:
refactored how settings files treated:
- created if doesn't exist
- config dirs added without using drupal's helpers. This was the problem before as it was trying to bootstrap Drupal before settings could be populated with DB credentials (from the next script).